json/codec.go

```diff
@@ -983,11 +986,11 @@ func syntaxError(b []byte, msg string, args ...interface{}) error {
 	return e
 }
 
-func inputError(b []byte, t reflect.Type) ([]byte, error) {
+func inputError(b []byte, t reflect.Type, flags inputFlags) ([]byte, error) {
```

Comment:
To reduce the need to explicitly pass the input flags as arguments to all these functions, we could make them methods of the `decoder` type.

Reply:
I thought that initially, but `inputError` is used outside the decoder:

```
json/parse.go
108:					b, err := inputError(b, t, flags)
141:					b, err := inputError(b, t, flags)
194:				b, err := inputError(b, t, flags)
```

Reply:
We could inline those particular cases?

Reply:
We could construct an empty decoder to call `inputError` on?

If there are only 3 call sites where we'd need to do this, it could be a good trade off to simplify the rest of the code.

Reply:
See a16b04ae6728bda418a15748992b375fd567745f.